.travis.yml

```
@@ -1,83 +1,95 @@
 # -*- coding: utf-8 -*-
 # vim: ft=yaml
 ---
+# Machine config
 dist: bionic
-stages:
-  - test
-  - lint
-  - name: release
-    if: branch = master AND type != pull_request
-
 sudo: required
-cache: bundler
-language: ruby
-
 services:
   - docker
 
-# Make sure the instances listed below match up with
-# the `platforms` defined in `kitchen.yml`
-# NOTE: Please try to select up to six instances that add some meaningful
-#       testing of the formula's behaviour. If possible, try to refrain from
-#       the classical "chosing all the instances because I want to test on
-#       another/all distro/s" trap: it will just add time to the testing (see
-#       the discussion on #121).  As an example, the set chosen below covers
-#       the most used distros families, systemd and non-systemd and the latest
-#       three supported Saltstack versions with python2 and 3.
-#       As for `kitchen.yml`, that should still contain all of the platforms,
-#       to allow for comprehensive local testing
-#       Ref: https://github.com/saltstack-formulas/template-formula/issues/118
-#       Ref: https://github.com/saltstack-formulas/template-formula/issues/121
-env:
-  matrix:
-    - INSTANCE: default-debian-10-develop-py3
-    # - INSTANCE: default-ubuntu-1804-develop-py3
-    # - INSTANCE: default-centos-7-develop-py3
-    # - INSTANCE: default-fedora-30-develop-py3
-    # - INSTANCE: default-opensuse-leap-15-develop-py3
-    # - INSTANCE: default-amazonlinux-2-develop-py2
-    # - INSTANCE: default-arch-base-latest-develop-py2
-    # - INSTANCE: default-debian-9-2019-2-py3
-    - INSTANCE: default-ubuntu-1804-2019-2-py3
-    # - INSTANCE: default-centos-7-2019-2-py3
-    # - INSTANCE: default-fedora-30-2019-2-py3
-    # - INSTANCE: default-opensuse-leap-15-2019-2-py3
-    - INSTANCE: default-amazonlinux-2-2019-2-py2
-    - INSTANCE: default-arch-base-latest-2019-2-py2
-    # - INSTANCE: default-debian-9-2018-3-py2
-    # - INSTANCE: default-ubuntu-1604-2018-3-py2
-    # - INSTANCE: default-centos-7-2018-3-py2
-    - INSTANCE: default-fedora-29-2018-3-py2
-    - INSTANCE: default-opensuse-leap-15-2018-3-py2
-    # - INSTANCE: default-amazonlinux-2-2018-3-py2
-    # - INSTANCE: default-arch-base-latest-2018-3-py2
-    # - INSTANCE: default-debian-8-2017-7-py2
-    # - INSTANCE: default-ubuntu-1604-2017-7-py2
-    - INSTANCE: centos6-centos-6-2017-7-py2
-    # - INSTANCE: default-fedora-29-2017-7-py2
-    # - INSTANCE: default-opensuse-leap-15-2017-7-py2
-    # - INSTANCE: default-amazonlinux-2-2017-7-py2
-    # - INSTANCE: default-arch-base-latest-2017-7-py2
+# Language and cache config
+language: ruby
+cache: bundler
 
+# Script to run for the test stage
 script:
   - bin/kitchen verify ${INSTANCE}
```

```suggestion
  - bin/kitchen verify '${INSTANCE}'
```
Adding single quotes here allows for REGEXP targeting in Kitchen, which I have a need for in `vault-formula` (without having to excessively escape the `INSTANCE` var, that is).

---

@dafyddj Ah, that's one of the points I was going to discuss with you in that PR!  While working on standardising our approach across formulas, the thought of using `kitchen list` as the basis for our matrix has crossed my mind many times.  The only thing that has stopped me is the excessive number of entries that can result from it, with `vault-formula` and `iptables-formula` being examples of that.  This wouldn't match up well with the regex method; it's definitely a clever method that needs some consideration (in terms of automation/maintenance).  I'd love to get more opinions about it.

---

This doesn't take anything away from the fact that quoting `${INSTANCE}` does absolutely no harm at all.  I'll definitely include this and we can continue this discussion in https://github.com/saltstack-formulas/vault-formula/pull/48.

---

Done.

---

Right, that hasn't worked.

https://travis-ci.com/saltstack-formulas/template-formula/jobs/243467344#L380-L383

```salt
$ bin/kitchen verify '${INSTANCE}'
-----> Starting Kitchen (v2.3.3)
>>>>>> 
No instances for regex `${INSTANCE}', try running `kitchen list'
```

So I've removed that commit and this will have to be covered in https://github.com/saltstack-formulas/vault-formula/pull/48.

---

Doh! I guess that should be double quotes, but yes, we can leave it for now